tests/test_kanboard.py

```diff
@@ -31,46 +33,75 @@
 class TestKanboard(unittest.TestCase):
 
     def setUp(self):
-        self.url = 'some api url'
+        self.url = 'http://someapiurl'
         self.client = client.Kanboard(self.url, 'username', 'password')
-        self.request, self.urlopen = self._create_mocks()
 
-    def test_api_call(self):
-        body = b'{"jsonrpc": "2.0", "result": true, "id": 123}'
-        self.urlopen.return_value.read.return_value = body
-        self.assertEquals(True, self.client.remote_procedure())
-        self.request.assert_called_once_with(self.url,
-                                             data=mock.ANY,
-                                             headers=mock.ANY)
+    # https://blog.sneawo.com/blog/2019/05/22/mock-aiohttp-request-in-unittests
+    def set_post_mock_response(self, mock_post, status, json_data):
+        mock_post.return_value.__aenter__.return_value.status = status
+        mock_post.return_value.__aenter__.return_value.json = asynctest.CoroutineMock(return_value=json_data)
 
-    def test_custom_auth_header(self):
+    def ignore_warnings(test_func):
+        def do_test(self, *args, **kwargs):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                test_func(self, *args, **kwargs)
+        return do_test
+
+    @asynctest.patch("aiohttp.ClientSession.post")
+    def test_api_call(self, mock_post):
+        body = {"jsonrpc": "2.0", "result": True, "id": 123}
+        self.set_post_mock_response(mock_post, 200, body)
+        result = self.client.remote_procedure()
+        self.assertEqual(True, result)
+        mock_post.assert_called_once_with(self.url,
+                                          data=mock.ANY,
+                                          headers=mock.ANY,
+                                          ssl=mock.ANY)
+
+    @asynctest.patch("aiohttp.ClientSession.post")
+    def test_custom_auth_header(self, mock_post):
         self.client._auth_header = 'X-Auth-Header'
-        body = b'{"jsonrpc": "2.0", "result": true, "id": 123}'
-        self.urlopen.return_value.read.return_value = body
-        self.assertEquals(True, self.client.remote_procedure())
-        self.request.assert_called_once()
-        _, kwargs = self.request.call_args
+        body = {"jsonrpc": "2.0", "result": True, "id": 123}
+        self.set_post_mock_response(mock_post, 200, body)
+        self.assertEqual(True, self.client.remote_procedure())
+        mock_post.assert_called_once()
```

Comment:
The Travis job failed with python 3.5 on this line because unittest.mock.Mock does not expose this method (introduced in python 3.6).

Maybe due to my cleaning the test dependencies. I'll look at it later.